File: maxRevenue.cpp
#include<iostream>
#include<bits/stdc++.h>
#include<array>
using namespace std;
int main() {
    int t;
    cin>>t;
    while (t!=0) {
        int n;
        long long result;
        cin>>n;
        array<int, > arr;
        for(int i = 0; i< n; i++) {
            cin>>arr[i];
        }
        for(int i = 0; i< n; i++) {
            cout<<arr[i];
        }
        cout<<endl;
        sort(arr.begin(), arr.end() , greater<int>());
        int maxRevenue, temp, tempRevenue, k = 1;
        maxRevenue = arr[0];
        for(int i = 0; i < n; i++) {
            temp = arr[i];
            tempRevenue = temp * k;
            if(tempRevenue > maxRevenue) {
                maxRevenue = tempRevenue;
                k += 1;
            }
            else {
                k += 1;
            }
        }
        cout<<maxRevenue<<endl;
        t -= 1;
    }
}

File: maxRevenue.py
n = int(input())
arr = []
for i in range(n):
    num = int(input())
    arr.append(num)

arr.sort(reverse = True)                   
maxRevenue = arr[0]                                         
i = 1
for val in range(0, len(arr)):
    tempCost = arr[val]
    tempRevenue = tempCost * i
    if tempRevenue > maxRevenue:
        maxRevenue = tempRevenue
        i += 1

    else:
        i += 1
print(maxRevenue)

